crates/bevy_core_pipeline/src/overlay/overlay.wgsl

```diff
@@ -0,0 +1,108 @@
+// numbers from https://www.shadertoy.com/view/lt3GRj
+
+struct VertexOutput {
+    [[builtin(position)]] clip_position: vec4<f32>;
+};
+
+struct Diagnostics {
+    fps: f32;
+};
+[[group(0), binding(0)]]
+var<uniform> diagnostics: Diagnostics;
+
+[[stage(vertex)]]
+fn vs_main(
+    [[builtin(vertex_index)]] in_vertex_index: u32,
+) -> VertexOutput {
+    var out: VertexOutput;
+    switch (in_vertex_index) {
+        case 0: { out.clip_position = vec4<f32>(-1.0, 3.0, 0.0, 1.0); }
+        case 1: { out.clip_position = vec4<f32>(-1.0, -1.0, 0.0, 1.0); }
+        case 2: { out.clip_position = vec4<f32>(3.0, -1.0, 0.0, 1.0); }
+        default: {}
+    }
+    return out;
+}
+
+fn digit_bin(x: i32) -> f32 {
+    switch (x) {
+        case 0: { return 480599.0; }
+        case 1: { return 139810.0; }
+        case 2: { return 476951.0; }
+        case 3: { return 476999.0; }
+        case 4: { return 350020.0; }
+        case 5: { return 464711.0; }
+        case 6: { return 464727.0; }
+        case 7: { return 476228.0; }
+        case 8: { return 481111.0; }
+        case 9: { return 481095.0; }
+        default: { return 0.0; }
+    }
+}
```

Comment:
Use a lookup table here.

Reply:
I feel like these are probably encoding some bitmask for the segments or something.